flink-dist/src/main/flink-bin/bin/config.sh

```diff
@@ -606,25 +606,33 @@ runBashJavaUtilsCmd() {
     local class_path=$2
     local conf_dir=$3
 
-    local output="`${JAVA_RUN} -classpath ${class_path} org.apache.flink.runtime.util.BashJavaUtils ${cmd} --configDir ${conf_dir} 2> /dev/null`"
+    local output=`${JAVA_RUN} -classpath ${class_path} org.apache.flink.runtime.util.BashJavaUtils ${cmd} --configDir ${conf_dir} 2> /dev/null | tail -n 1`
```

Comment:
If we remove `2> /dev/null`, log will pollute the output in the `testGetTmResourceDynamicConfigs` test.

Reply:
Shouldn't we also take the last line of the outputs out in `BashJavaUtilsITCase`?

Reply:
I think we could ignore the error stream in `JavaBashTestBase#executeScript`

Reply:
`JavaBashTest` is a common purpose util class. I would not ignore the error stream in it, because ignoring the error stream is not a common contract.
Maybe we can override `executeScript` in `BashJavaUtilsITCase` and fetch the last line of the outputs there, because putting the results in the last line is the contract of `BashJavaUtils`.

Reply:
In addition, I think we should add another test case to validate the `BashJavaUtils` works well when there are other noises in the outputs.

Reply:
If so, I prefer to add a new function called `executeScriptAndFetchLastLine`. It's a bit confusing to only return the last line from `executeScript `.

Reply:
Sounds good.
We can also specify `RUN_BASH_JAVA_UTILS_CMD_SCRIPT` in `executeScriptAndFetchLastLine`, leaving `GET_TM_RESOURCE_DYNAMIC_CONFIGS/GET_TM_RESOURCE_JVM_PARAMS` the only argument of it.

Reply:
> In addition, I think we should add another test case to validate the `BashJavaUtils` works well when there are other noises in the outputs.

I don't think `BashJavaUtils` need to tolerant the noises in the output. The noises are filter by `runBashJavaUtilsCmd`.

Reply:
True, that's what I mean that should be validated. Sorry for the confusion.